strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/BrowseController.java

```diff
@@ -0,0 +1,163 @@
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+
+/**
+ * REST API for browsing storage/repository/filesystem structures.
+ *
+ * @author Guido Grazioli
+ */
+@RestController
+@RequestMapping(path = BrowseController.ROOT_CONTEXT)
+public class BrowseController
+        extends BaseArtifactController
+{
+
+    private static final Logger logger = LoggerFactory.getLogger(BrowseController.class);
+
+    // must be the same as @RequestMapping value on the class definition
+    public final static String ROOT_CONTEXT = "/browse";
+
+    @ApiOperation(value = "List configured storages.")
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "The list was returned."),
+                            @ApiResponse(code = 400, message = "An error occurred.") })
+    @PreAuthorize("hasAuthority('ARTIFACTS_RESOLVE')")
+    @RequestMapping(method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<String> browse()
+    {
+        try
+        {
+            Map<String,SortedSet<String>> storages = new HashMap<>();
+            storages.put("storages", new TreeSet<String>(configurationManager.getConfiguration()
+                    .getStorages().keySet()));
+            return ResponseEntity.ok(new ObjectMapper().writeValueAsString(storages));
+        }
+        catch (Exception e)
+        {
+            logger.error(e.getMessage(), e);
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR)
+                    .body(String.format("{ 'error': '%s' }", e.getMessage()));
+        }
+    }
+    
+    @ApiOperation(value = "List configured repositories for a storage.")
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "The list was returned."),
+                            @ApiResponse(code = 404, message = "The requested storage was not found."),
+                            @ApiResponse(code = 400, message = "An error occurred.") })
+    @PreAuthorize("hasAuthority('ARTIFACTS_RESOLVE')")
+    @RequestMapping(value="/{storageId}", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<String> repositories(@ApiParam(value = "The storageId", required = true)
+                                               @PathVariable("storageId") String storageId)
+    {
+        try
+        {
+            Storage storage = configurationManager.getConfiguration().getStorage(storageId);
+            if (storage == null) 
+            {
+                return ResponseEntity.status(HttpStatus.NOT_FOUND)
+                        .body("{ 'error': 'The requested storage was not found.' }");
```

Comment:
@guidograzioli : Could you please re-indent this according to the [coding convention](https://github.com/strongbox/strongbox/wiki/Coding-Convention)? Thanks! :)